android/source/VideoLocker/src/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -0,0 +1,213 @@
+                    optional_text.setText(getString(R.string.hide_optional_text));
+                }
+                else{
+                    optionalFieldsLayout.setVisibility(v.GONE);
+                    optional_text.setText(getString(R.string.show_optional_text));
+                }
+            }
+        });
+
+        ImageButton closeButton = (ImageButton) findViewById(R.id.actionbar_close_btn);
+        if(closeButton!=null){
+            closeButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    finish();
+                }
+            });
+        }
+
+        ETextView customTitle = (ETextView) findViewById(R.id.activity_title);
+        if(customTitle!=null){
+            customTitle.setText(getString(R.string.register_title));
+        }
+
+        ETextView eulaTv = (ETextView) findViewById(R.id.end_user_agreement_tv);
+        eulaTv.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showEulaDialog();
+            }
+        });
+
+        setupRegistrationForm();
+
+        hideSoftKeypad();
+    }
+
+    public void showEulaDialog() {
+        showWebDialog(getString(R.string.eula_file_link), true,
+                getString(R.string.end_user_title));
+    }
+
+    @Override
+    public void finish() {
+        super.finish();
+        overridePendingTransition(R.anim.no_transition, R.anim.slide_out_to_bottom);
+    }
+
+    //Enable all the Fields(Views) which were disabled
+    private void setElementsEnabled(){
+        createButtonEnabled();
+    }
+
+    //Disable the Fields(Views) when doing server call
+    private void setElementsDisabled(){
+        createButtonDisabled();
+    }
+
+    //Disable the Create button during server call
+    private void createButtonDisabled() {
+        createAccountBtn.setBackgroundResource(R.drawable.new_bt_signin_active);
+        createAccountBtn.setEnabled(false);
+        createAccountTv.setText(getString(R.string.creating_account_text));
+    }
+
+    //Enable the Create button during server call
+    private void createButtonEnabled() {
+        createAccountBtn.setBackgroundResource(R.drawable.bt_signin_active);
+        createAccountBtn.setEnabled(true);
+        createAccountTv.setText(getString(R.string.create_account_text));
+    }
+
+    private void setupRegistrationForm() {
+        try {
+            InputStream in = getAssets().open("config/registration_form.json");
+            Gson gson = new Gson();
+            RegistrationForm form = gson.fromJson(new InputStreamReader(in), RegistrationForm.class);
+
+            LayoutInflater inflater = getLayoutInflater();
+
+            for (RegistrationFormField field : form.getFields()) {
+                IRegistrationFieldView fieldView = null;
+
+                if (field.getFieldType().equalsIgnoreCase(RegistrationFieldType.EMAIL.toString())
```

Comment:
This can be simplified a bit if we use the enum more directly as described above.